[AI-5206] DDS: WatchGuard Firebox Integration v1.0.0 (#20030)
DataDog/integrations-core: watchguard_firebox/README.md

## Overview

> [WatchGuard Firebox][4] is a network security platform offering comprehensive threat protection, including firewall, networking and other security features, suitable for various network environments, from small offices to large enterprises.

Reviewer: ```suggestion
[WatchGuard Firebox][4] is a network security platform offering comprehensive threat protection, including firewall, networking, and other security features, suitable for various network environments, from small offices to large enterprises.
```

Author: Have updated as suggested.

- **Networking Events**: Represents log messages generated for traffic related to the connections through your Firebox. This can include events related to interface activity, dynamic routing, PPPoE connections, and DHCP server requests.
- **Management Events**: Represents log messages generated for activity on your Firebox. This includes when changes are made to the device configuration and Device Management user accounts, for user authentication to the Firebox, and actions related to system settings.
- **VPN Events**: Represents log messages generated for processes related to the all VPNs configured on your Firebox. This includes changes to the VPN configuration, tunnel status, and daemon activity.
- **Proxy Policy Events**: Represents log messages generated for traffic managed by the proxy policies configured on your Firebox. This can include events related to traffic through the proxy, proxy actions, and security services.

> This integration seamlessly collects all the above listed logs, channeling them into Datadog for analysis. Leveraging the built-in logs pipeline, these logs are parsed and enriched, enabling effortless search and analysis. The integration provides insight into firewall, networking, management, vpn and proxy policy events through the out-of-the-box dashboards. Additionally, it includes ready-to-use Cloud SIEM detection rules for enhanced monitoring and security.

Reviewer: ```suggestion
This integration seamlessly collects all the above listed logs, channeling them into Datadog for analysis. Leveraging the built-in logs pipeline, these logs are parsed and enriched, enabling effortless search and analysis. The integration provides insight into firewall, networking, management, VPN, and proxy policy events through the out-of-the-box dashboards. Additionally, it includes ready-to-use Cloud SIEM detection rules for enhanced monitoring and security.
```

## Setup

### Installation

To install the WatchGuard Firebox integration, run the following Agent installation command and the steps below. For more information, see the [Integration Management][5] documentation.

**Note**: This step is not necessary for Agent version >= 7.66.0.

Linux command:

  ```shell
  sudo -u dd-agent -- datadog-agent integration install datadog-watchguard-firebox==1.0.0
  ```

### Configuration

#### Log collection

1. Collecting logs is disabled by default in the Datadog Agent. Enable it in `datadog.yaml`:

    ```yaml
    logs_enabled: true
    ```

2. Add this configuration block to your `watchguard-firebox.d/conf.yaml` file to start collecting your WatchGuard Firebox logs. See the sample [watchguard-firebox.d/conf.yaml][6] for available configuration options.

      ```yaml
      logs:
       - type: udp
         port: <PORT>
         service: watchguard-firebox
         source: watchguard-firebox
      ```

      Note:

>       - PORT: Port should be similar to the port provided in **Configure Syslog Message Forwarding from WatchGuard Firebox Server**.

Reviewer: Should the port be similar (example: the port 5001 is similar to the port 5000), or identical?

Author: Understood, have changed similar to identical

      - It is recommended not to change the service and source values, as these parameters are integral to the pipeline's operation.

3. [Restart the Agent][1].

#### Configure Syslog Message Forwarding from WatchGuard Firebox Server

1. Log in to **Fireware Web UI** with an administrator account.
2. Select **System** > **Logging**.
3. Select the **Syslog Server** tab.
4. Select the **Send log messages to these syslog servers** check box.

> 5. Click **Add**, then the Syslog Server dialog box opens.

Reviewer: ```suggestion
5. Click **Add** to open the Syslog Server dialog box.
```

6. In the **IP Address** text box, enter the IP address of the server on which datadog-agent is installed.
7. In the **Port** text box, enter the port you configured to receive the Firebox logs.
8. From the **Log Format** drop-down list, select **Syslog**.
9. Select **The time stamp** and **The serial number of the device** check boxes.
10. Click **OK**.

### Validation

[Run the Agent's status subcommand][2] and look for `watchguard-firebox` under the Checks section.

## Data Collected

### Logs

> The WatchGuard Firebox integration collects Firewall, Networking, Management, VPN and Proxy Policy Events.

Reviewer: ```suggestion
The WatchGuard Firebox integration collects firewall, networking, management, VPN, and proxy policy Events.
```
Just because these weren't capitalized earlier in a similar sentence.

### Events

The WatchGuard Firebox integration does not include any events.

## Troubleshooting

### WatchGuard Firebox

**Permission denied while port binding:**

If you see a **Permission denied** error while port binding in the Agent logs, see the following instructions:

   1. Binding to a port number under 1024 requires elevated permissions. Grant access to the port using the `setcap` command:

      - Grant access to the port using the `setcap` command:

         ```shell
         sudo setcap CAP_NET_BIND_SERVICE=+ep /opt/datadog-agent/bin/agent/agent
         ```

      - Verify the setup is correct by running the `getcap` command:

         ```shell
         sudo getcap /opt/datadog-agent/bin/agent/agent
         ```

         With the expected output:

         ```shell
         /opt/datadog-agent/bin/agent/agent = cap_net_bind_service+ep
         ```

         **Note**: Re-run this `setcap` command every time you upgrade the Agent.

   2. [Restart the Agent][1].

**Data is not being collected:**

Make sure that traffic is bypassed from the configured port if the firewall is enabled.

**Port already in use:**

> If you see the **Port <PORT-NO\> Already in Use** error, see the following instructions. The example below is for PORT-NO = 514:

Reviewer: ```suggestion
If you see the **Port <PORT_NUMBER> Already in Use** error, see the following instructions. The example below is for a PORT_NUMBER equal to 514:
```

Author: Have updated as suggested.
